test/tests/bootstrap/test_api20_esxi_bootstrap.py

```diff
@@ -0,0 +1,128 @@
+'''
+Copyright 2017 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+This script tests arbitrary payload of the RackHD API 2.0 OS bootstrap workflows.
+The default case is running a minimum payload ESXi 6.0 OS install.
+Other ESXi OS install cases can be specified by creating a payload file and specifiying it using the '-extra' argument.
+This test takes 15-20 minutes to run.
+
+Example payload file (installed in configuration dir):
+
+{"bootstrap-payload":
+    {"name": "Graph.InstallESXi",
+        "options": {"defaults": {
+        "version": "6",
+        "repo": "http://172.31.128.1:8080/ESXi/6.0",
+        "rootPassword": "1234567",
+        "hostname": "rackhdnode",
+        "users": [{"name": "rackhduser",
+                   "password": "RackHDRocks!",
+                   "uid": 1010}]}}}
+}
+
+Example command line using external payload file:
+
+python run_tests.py -stack 4 -test tests/bootstrap/test_api20_esxi_bootstrap.py -extra base_esxi_55_install.json
+
+'''
+
+import fit_path  # NOQA: unused import
+from nose.plugins.attrib import attr
+import fit_common
+import flogging
+import random
+import time
+from nosedep import depends
+log = flogging.get_loggers()
+
+# sample default base payload
+PAYLOAD = {"name": "Graph.InstallESXi",
+           "options": {"defaults": {"version": "6",
+                                    "repo": "http://172.31.128.1:8080/ESXi/6.0",
+                                    "rootPassword": "1234567",
+                                    "hostname": "rackhdnode",
+                                    "users": [{"name": "rackhduser",
+                                               "password": "RackHDRocks!",
+                                               "uid": 1010}]}}}
+# if an external payload file is specified, use that
+config = fit_common.fitcfg().get('bootstrap-payload', None)
+if config:
+    PAYLOAD = config
+
+
+# this routine polls a workflow task ID for completion
+def wait_for_workflow_complete(instanceid, start_time, waittime=900, cycle=30):
+    log.info_5(" Workflow started at time: " + str(start_time))
+    while start_time - time.time() < waittime:  # limit test to 30 minutes
+        result = fit_common.rackhdapi("/api/2.0/workflows/" + instanceid)
+        if result['status'] != 200:
+            log.error(" HTTP error: " + result['text'])
+            return False
+        if result['json']['status'] in ['running', 'pending']:
+            log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
```

Comment:
Flogger messages should not use format or " + ". Standard "C" formatting is to be used. I'm not going to hold up the PR for this though.